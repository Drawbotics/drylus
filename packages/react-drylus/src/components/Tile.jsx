import sv from '@drawbotics/drylus-style-vars';
import { css, cx } from 'emotion';
import PropTypes from 'prop-types';
import React from 'react';

const styles = {
  root: css`
    background: ${sv.white};
    box-shadow: ${sv.elevation1};
    padding: ${sv.defaultPadding};

    @media ${sv.screenL} {
      padding: ${sv.paddingSmall};
    }
  `,
  title: css`
    color: ${sv.colorSecondary};
    text-transform: uppercase;
    font-weight: 500;
    font-size: 0.9rem;
    margin-bottom: ${sv.marginSmall};

    @media ${sv.screenL} {
      margin-bottom: ${sv.marginExtraSmall};
    }
  `,
  noPadding: css`
    padding: 0px !important;
  `,
};

const Tile = ({ style, title, children, noPadding }) => {
  return (
    <div
      style={style}
      className={cx(styles.root, {
        [styles.noPadding]: noPadding,
      })}>
      {do {
        if (title != null) {
          <div className={styles.title}>{title}</div>;
        }
      }}
      {children}
    </div>
  );
};

Tile.propTypes = {
  /** Displayed at the top of the tile, looks like a label */
  title: PropTypes.string,

  /** Content displayed within the tile */
  children: PropTypes.node.isRequired,

  /** Used for style overrides */
  style: PropTypes.object,

  /** Removes the space between the content and the borders (good for full size images in the tile) */
  noPadding: PropTypes.bool,
};

export default Tile;
