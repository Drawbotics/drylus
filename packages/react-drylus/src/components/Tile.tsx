import sv from '@drawbotics/drylus-style-vars';
import { css, cx } from 'emotion';
import React from 'react';

import { Style } from '../types';
import { run } from '../utils';

const styles = {
  root: css`
    background: ${sv.white};
    box-shadow: ${sv.elevation1};
    padding: ${sv.defaultPadding};

    @media ${sv.screenL} {
      padding: ${sv.paddingSmall};
    }
  `,
  title: css`
    color: ${sv.colorSecondary};
    text-transform: uppercase;
    font-weight: 500;
    font-size: 0.9rem;
    margin-bottom: ${sv.marginSmall};

    @media ${sv.screenL} {
      margin-bottom: ${sv.marginExtraSmall};
    }
  `,
  noPadding: css`
    padding: 0px !important;
  `,
};

export interface TileProps {
  /** Displayed at the top of the tile, looks like a label */
  title?: string;

  /** Content displayed within the tile */
  children: React.ReactNode;

  /** Removes the space between the content and the borders (good for full size images in the tile) */
  noPadding?: boolean;

  /** Used for style overrides */
  style?: Style;

  /** Used for style overrides */
  className?: string;
}

export const Tile = ({ style, title, children, noPadding, className }: TileProps) => {
  return (
    <div
      style={style}
      className={cx(
        styles.root,
        {
          [styles.noPadding]: noPadding === true,
        },
        className,
      )}>
      {run(() => {
        if (title != null) {
          return <div className={styles.title}>{title}</div>;
        }
      })}
      {children}
    </div>
  );
};
