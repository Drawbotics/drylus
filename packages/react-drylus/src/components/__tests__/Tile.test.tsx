import React from 'react';
import { create } from 'react-test-renderer';

import { Tile } from '../Tile';

describe('Tile', () => {
  describe('matches snapshot when', () => {
    it('has a title', () => {
      const tree = create(<Tile title="Title">Tile content</Tile>).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('does not have a title', () => {
      const tree = create(<Tile>Tile content</Tile>).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('does not have any padding', () => {
      const tree = create(<Tile noPadding>Tile content</Tile>).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
