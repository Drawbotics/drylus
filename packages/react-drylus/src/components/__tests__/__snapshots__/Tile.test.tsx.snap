// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tile matches snapshot when does not have a title 1`] = `
.emotion-0 {
  background: #fff;
  box-shadow: 0 2px 10px rgba(135,155,166,0.1);
  padding: 24px;
}

@media only screen and (max-width:768px) {
  .emotion-0 {
    padding: 16px;
  }
}

<div
  className="emotion-0"
>
  Tile content
</div>
`;

exports[`Tile matches snapshot when does not have any padding 1`] = `
.emotion-0 {
  background: #fff;
  box-shadow: 0 2px 10px rgba(135,155,166,0.1);
  padding: 24px;
  padding: 0px !important;
}

@media only screen and (max-width:768px) {
  .emotion-0 {
    padding: 16px;
  }
}

<div
  className="emotion-0"
>
  Tile content
</div>
`;

exports[`Tile matches snapshot when has a title 1`] = `
.emotion-1 {
  background: #fff;
  box-shadow: 0 2px 10px rgba(135,155,166,0.1);
  padding: 24px;
}

@media only screen and (max-width:768px) {
  .emotion-1 {
    padding: 16px;
  }
}

.emotion-0 {
  color: #879ba6;
  text-transform: uppercase;
  font-weight: 500;
  font-size: 0.9em;
  margin-bottom: 16px;
}

@media only screen and (max-width:768px) {
  .emotion-0 {
    margin-bottom: 8px;
  }
}

<div
  className="emotion-1"
>
  <div
    className="emotion-0"
  >
    Title
  </div>
  Tile content
</div>
`;
